Marlin/src/module/probe.cpp

```diff
@@ -919,11 +919,6 @@ float Probe::probe_at_point(const_float_t rx, const_float_t ry, const ProbePtRai
     DEBUG_POS("", current_position);
   }
 
-  #if ENABLED(BLTOUCH)
-    // Reset a BLTouch in HS mode if already triggered
-    if (bltouch.high_speed_mode && bltouch.triggered()) bltouch._reset();
-  #endif
-
```

Comment:
We should check in with the author of #16811 — @InsanityAutomation — regarding removing this and moving it down below, where it may never be reached due to some intermediate `return` statements. This is meant to catch the probe being triggered before any move is done at all, which may be due to any cause, including a call to `M401` before a call to `G30` or something like that.

Reply:
The way I read #16811 solves "an early trigger keeping raising Z", by
two key changes, reset probe if already triggering (which will clear
some issues), and after run_z_probe() where it previously would always
raise, after #16811 it will only raise if the measurement is not NaN.

I'll relocate the reset to to be right after the do_blocking_move_to
which moves one one return statement earlier.  That leaves one return
at the start of probe_at_point that can do an early return if
can_reach is false, but that return is prior to any movements.  I
don't expect this to cause any problems for #16811.

Doing the trigger test before the raise Z is giving me problem on G29
with in high speed mode with some other optimizations.  What I was
finding was
1. probe is deployed
2. `BLTouch from 90 to 10`
3. it makes two samples at the first location
4. probe_at_point() is called for the next point
4.1. the probe is still triggered
5. `bltouch._reset();`
5.1. `BLTouch from 10 to 160`
6. fails with `SLOW Probe fail! - No trigger.`
7. the command was never given to deploy.
I was looking at the M111 S32 output.

I'd need to do additional debugging to figure out why `deploy()` right
before `run_z_probe(` didn't switch it to S10.  I have manually
deployed the probe, put it in an alarm condition, then executed G30,
which reset the BLTouch and gave a good G30 value, so I know the
_reset()/deploy() can clear an alarm condition.

I see you're making and pushing changes, how would you like me to make
my updates available without causing issues for you?  Put my changes
on the tip of dfries:bugfix-2.1.x_probe associated with #25652?

Reply:
Just do the usual commits. I'm just merging the latest changes which add merge commits, so you just need to pull before making new commits.